fs/entry.go

```diff
@@ -78,6 +79,22 @@ type ErrorEntry interface {
 // ErrEntryNotFound is returned when an entry is not found.
 var ErrEntryNotFound = errors.New("entry not found")
 
+// ReaddirToIterate is an adapter for a naive Readdir -> IterateEntries implementation.
+func ReaddirToIterate(ctx context.Context, d Directory, cb func(context.Context, Entry) error) error {
```

Comment:
can we add a simple unit test here? other than that LGTM.

Reply:
is there another package that this function would be a good fit for or a good `Directory` implementation in which to place the test? Unfortunately even a simple test in `entries_test.go` would require minimal mocks for the `Directory`, `Entry`, and `os.FileInfo` interfaces as I cannot import `internal/mockfs` or any other concrete implementations of `Directory` due to import cycles